iceoryx_posh/include/iceoryx_posh/roudi/port_pool.hpp

```diff
@@ -49,39 +55,60 @@ class PortPool
     /// @todo don't create the vector with each call but only when the data really change
     /// there could be a member "cxx::vector<popo::SenderPortData* m_senderPorts;" and senderPorts() would just update
     /// this member if the sender ports actually changed
+    /// @deprecated #25
     virtual cxx::vector<SenderPortType::MemberType_t*, MAX_PORT_NUMBER> senderPortDataList() noexcept = 0;
+    /// @deprecated #25
     virtual cxx::vector<ReceiverPortType::MemberType_t*, MAX_PORT_NUMBER> receiverPortDataList() noexcept = 0;
-    cxx::vector<popo::InterfacePortData*, MAX_INTERFACE_NUMBER> interfacePortDataList() noexcept;
-    cxx::vector<popo::ApplicationPortData*, MAX_PROCESS_NUMBER> appliactionPortDataList() noexcept;
-    cxx::vector<runtime::RunnableData*, MAX_RUNNABLE_NUMBER> runnableDataList() noexcept;
-
+    virtual cxx::vector<PublisherPortRouDiType::MemberType_t*, MAX_PORT_NUMBER> getPublisherPortDataList() noexcept = 0;
+    virtual cxx::vector<SubscriberPortProducerType::MemberType_t*, MAX_PORT_NUMBER>
+    getSubscriberPortDataList() noexcept = 0;
+    cxx::vector<popo::InterfacePortData*, MAX_INTERFACE_NUMBER> getInterfacePortDataList() noexcept;
+    cxx::vector<popo::ApplicationPortData*, MAX_PROCESS_NUMBER> getApplicationPortDataList() noexcept;
+    cxx::vector<runtime::RunnableData*, MAX_RUNNABLE_NUMBER> getRunnableDataList() noexcept;
+
+    /// @deprecated #25
     virtual cxx::expected<SenderPortType::MemberType_t*, PortPoolError>
     addSenderPort(const capro::ServiceDescription& serviceDescription,
                   mepoo::MemoryManager* const memoryManager,
                   const std::string& applicationName,
                   const mepoo::MemoryInfo& memoryInfo = mepoo::MemoryInfo()) noexcept = 0;
 
+    /// @deprecated #25
     virtual cxx::expected<ReceiverPortType::MemberType_t*, PortPoolError>
     addReceiverPort(const capro::ServiceDescription& serviceDescription,
                     const std::string& applicationName,
                     const mepoo::MemoryInfo& memoryInfo = mepoo::MemoryInfo()) noexcept = 0;
 
+    virtual cxx::expected<PublisherPortRouDiType::MemberType_t*, PortPoolError>
```

Comment:
I think there's no need to make this method `virtual`, same for the other one

Reply:
missed